edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/PersistedTokenCredentialsCache.cs

```diff
@@ -22,10 +23,13 @@ public async Task Add(IClientCredentials clientCredentials)
         {
             if (clientCredentials is ITokenCredentials tokenCredentials)
             {
+                string targetId = AuthChainHelpers.GetAuthTarget(tokenCredentials.AuthChain).GetOrElse(tokenCredentials.Identity.Id);
+
                 try
                 {
-                    var tokenCredentialsData = new TokenCredentialsData(tokenCredentials.Token, tokenCredentials.IsUpdatable);
-                    await this.encryptedStore.Put(tokenCredentials.Identity.Id, tokenCredentialsData.ToJson());
+                    var tokenCredentialsData = new TokenCredentialsData(tokenCredentials.Token, tokenCredentials.IsUpdatable, tokenCredentials.AuthChain);
+                    string tokenCredentialsString = JsonConvert.SerializeObject(tokenCredentialsData);
```

Comment:
Why does toJson not work here?

Reply:
there's a definition ambiguity where ToJson() is declare in Newtonsoft.Json and Edge.Util.Json, this is just my work around for it.